### PR TITLE
scraper: various tweaks and fixes

### PR DESCRIPTION
**Pull request by raymondng**
- handle some exceptions
- config tweaks
- timingWithResult
- better handling of states (esp wrt SCRAPE_FAILED, UNSCRAPABLE)

cc: @ymatsuda

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/scraper/ScraperConfig.scala [hunk 1]
```diff
@@ -5,22 +5,22 @@ import scala.util.Random
 case class ScraperConfig(
   minInterval: Double = 24.0d, //hours
   maxInterval: Double = 1024.0d, //hours
-  intervalIncrement: Double = 4.0d, //hours
+  intervalIncrement: Double = 6.0d, //hours
   intervalDecrement: Double = 2.0d, //hours
-  initialBackoff: Double = 2.0d, //hours
+  initialBackoff: Double = 12.0d, //hours
```
Comment (ymatsuda) [inline kifi-backend/modules/common/app/com/keepit/scraper/ScraperConfig.scala:10]: A half day seems very long. Is the retry actually a problem?

File: kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala [hunk 2]
```diff
@@ -13,6 +13,7 @@ import play.api.libs.concurrent.Execution.Implicits.defaultContext
 import play.modules.statsd.api.Statsd
 import org.feijoas.mango.common.cache._
 import java.util.concurrent.TimeUnit
+import com.keepit.scraper.ScraperConfig
```
Comment (ymatsuda) [inline kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala:16]: ?

File: kifi-backend/modules/common/app/com/keepit/scraper/ScraperConfig.scala (hunk 1 above)
Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/scraper/ScraperConfig.scala:10]: It's to be on the conservative side. Given it already failed (including stuck ones) I feel that we shouldn't be retrying too soon ... but, I do admit the # is chosen not scientifically. WDYT?

File: kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala [hunk 3]
```diff
@@ -115,6 +116,12 @@ extends DbRepo[NormalizedURI] with NormalizedURIRepo with ExternalIdColumnDbFunc
           scrapeRepo.save(scrapeInfo.withStateAndNextScrape(ScrapeInfoStates.INACTIVE))
         case _ => // do nothing
       }
+    } else if (uri.state == NormalizedURIStates.SCRAPE_FAILED) {
+      scrapeRepo.getByUriId(saved.id.get) match { // do not use saveStateAndNextScrape
+        case Some(scrapeInfo) => if (scrapeInfo.state == ScrapeInfoStates.INACTIVE) { scrapeRepo.save(scrapeInfo.withState(ScrapeInfoStates.ACTIVE)) }
+        case Some(scrapeInfo) => // do nothing
+        case None => scrapeRepo.save(ScrapeInfo(uriId = saved.id.get))
```
Comment (ymatsuda) [inline kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala:123]: Why setting uriId and save here? Is there any side effect we want?

File: kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala (hunk 2 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala:16]: Good catch :)

Context: the current code (before this change; line 122) uses the old withStateAndNextScrape() which actually sets next_scrape to START_OF_TIME, which IMO is probably not the right thing to do wrt SCRAPE_FAILED cases. 

So I thought that if we were to update scrapeInfo at all for the SCRAPE_FAILED cases, we may as well make sure that it's updated withFailure (and that requires ScraperConfig).

However I backed off from that as it introduces extra dependency -- so I settled with merely making sure we don't set next_scrape to START_OF_TIME.

I'll look into this some more while refactoring the code but for now I will remove this import.

File: kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala (hunk 3 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala:123]: Thanks for mentioning this -- I was actually wondering about this one as well.

This line is "inherited" from the current code (line 125/132) so the behavior has always been there -- having said that, I'd be more than happy to remove it for SCRAPE_FAILED case.

File: kifi-backend/modules/common/app/com/keepit/scraper/ScraperConfig.scala (hunk 1 above)
Comment (ymatsuda) [inline kifi-backend/modules/common/app/com/keepit/scraper/ScraperConfig.scala:10]: We use an exponential backoff. You are not saving too much. I think it is better retry is the failure is due to  temporary outage. I expect such an outage usually recovers in a few hours.

**Pull request merged**